test/byContract/ENSIntegration/ENSIntegration.ts

```diff
@@ -0,0 +1,128 @@
+            expect(await resolve('datasets.v5.iexec.eth')).to.equal(
+                await iexecPoco.datasetregistry(),
+            );
+            expect(await resolve('workerpools.v5.iexec.eth')).to.equal(
+                await iexecPoco.workerpoolregistry(),
+            );
+        });
+    });
+
+    describe('Reverse resolution', async () => {
+        it('Should lookup initial addresses', async () => {
+            expect(await lookup(iexecAdmin.address)).to.equal('admin.iexec.eth');
+            expect(await lookup(proxyAddress)).to.equal('core.v5.iexec.eth');
+            expect(await lookup(await iexecPoco.appregistry())).to.equal('apps.v5.iexec.eth');
+            expect(await lookup(await iexecPoco.datasetregistry())).to.equal(
+                'datasets.v5.iexec.eth',
+            );
+            expect(await lookup(await iexecPoco.workerpoolregistry())).to.equal(
+                'workerpools.v5.iexec.eth',
+            );
+        });
+
+        it('Should register reverse resolution name', async () => {
+            const name = 'test.domain.eth';
+            const reverseNameHash = ethers.utils.namehash(
+                `${proxyAddress.substring(2)}.addr.reverse`,
+            );
+            const reverseRootNameHash = ethers.utils.namehash('addr.reverse');
+            const reverseRegistrarAddress = await ensRegistry.owner(reverseRootNameHash);
+            const reverseResolverAddress = await ReverseRegistrar__factory.connect(
+                reverseRegistrarAddress,
+                anyone,
+            ).defaultResolver();
+            const reverseResolver = PublicResolver__factory.connect(reverseResolverAddress, anyone);
+            await expect(iexecPocoAsAdmin.setName(ensRegistry.address, name))
+                .to.emit(reverseResolver, 'NameChanged')
+                .withArgs(reverseNameHash, name);
+            expect(await lookup(proxyAddress)).to.equal(name);
+        });
+
+        it('Should not register reverse resolution name when sender is not the owner', async () => {
+            await expect(
+                iexecPoco.setName(ensRegistry.address, 'some.name.eth'),
+            ).to.be.revertedWith('Ownable: caller is not the owner');
+        });
+    });
+
+    /**
+     * Get the address associated to the given ENS name using forward resolution.
+     * @param domain ENS domain name
+     * @returns ETH address
+     */
+    async function resolve(domain: string) {
+        const nameHash = ethers.utils.namehash(domain);
+        const resolver = await getResolver(nameHash);
+        return await resolver['addr(bytes32)'](nameHash);
+    }
+
+    /**
+     * Get the primary ENS name associated to the given address using reverse resolution.
+     * @param address
+     * @returns ENS name
+     */
+    async function lookup(address: string) {
+        const nameHash = ethers.utils.namehash(`${address.substring(2)}.addr.reverse`);
+        const reverseResolver = await getResolver(nameHash);
```

Comment:
what is the difference between reverseResolver and reverse line 104

Reply:
Both are resovers, I specified `reverseResolver` to emphasize that it's used to do reverse resolving.